base/pkg/entry.jl

```diff
@@ -48,25 +48,23 @@ end
 
 function add(pkg::AbstractString, vers::VersionSet)
     outdated = :maybe
-    @sync begin
-        @async if !edit(Reqs.add,pkg,vers)
-            ispath(pkg) || throw(PkgError("unknown package $pkg"))
-            info("Nothing to be done")
-        end
-        branch = Dir.getmetabranch()
-        outdated = with(GitRepo, "METADATA") do repo
-            if LibGit2.branch(repo) == branch
-                if LibGit2.isdiff(repo, "origin/$branch")
-                    outdated = :yes
-                else
-                    try
-                        LibGit2.fetch(repo)
-                        outdated = LibGit2.isdiff(repo, "origin/$branch") ? (:yes) : (:no)
-                    end
-                end
+    if !edit(Reqs.add,pkg,vers)
+        ispath_casesensitive(pkg) || throw(PkgError("unknown package $pkg"))
+        info("Nothing to be done")
```

Comment:
The message here seems a bit misleading.

If I do `Pkg.add("Foo")` and I already have a directory `Foo` in `Pkg.dir()`, but `Foo` is not the `Foo` repo corresponding to the one in METADATA, I would expect an error rather than "Nothing to be done".   Should we check that their origins match?

And if I already have a directory `foo`, an "unknown package Foo" error message is wrong.

At the very least, we should probably be doing:
```jl
ispath(pkg) || throw(PkgError("unknown package $pkg"))
Read.isgitrepo(pkg) || throw(PkgError("Cannot install $pkg because there is a pre-existing $pkg directory in $(pwd())"))
```